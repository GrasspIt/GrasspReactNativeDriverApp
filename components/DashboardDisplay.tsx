import React, { useRef, useState } from 'react';
import { SafeAreaView, StyleSheet, Text, View } from 'react-native';
import { ActivityIndicator, Button, useTheme } from 'react-native-paper';
import { StatusBar } from 'expo-status-bar';

import OnCallSwitch from './OnCallSwitch';
import AlertSuccessOrError from "./AlertSuccessOrError";

//tests
import { AutocompleteDropdown } from "react-native-autocomplete-dropdown"

type Props = {
  dspr;
  dsprDriver;
  isLoading: boolean;
  getDriverData;
  setDriverOnCallState;
  showLocationPermissionAlert: boolean;
  closeLocationPermissionAlert: () => any;
  locationPermissionAlertTitle: string;
  locationPermissionAlertText: string;
};

const DashboardDisplay = ({
  dspr,
  dsprDriver,
  isLoading,
  getDriverData,
  setDriverOnCallState,
  showLocationPermissionAlert,
  closeLocationPermissionAlert,
  locationPermissionAlertTitle,
  locationPermissionAlertText
}: Props) => {
  const { colors } = useTheme();

  return (
    <SafeAreaView style={{ flex: 1 }}>
      {isLoading ? (
        <View style={[styles.container, { backgroundColor: colors.background }]}>
          <ActivityIndicator size='large' color={colors.primary} />
        </View>
      ) : dsprDriver ? (
        <View style={{ flex: 1, backgroundColor: colors.background }}>
          <Text style={styles.dsprTitle}>{dspr.name}</Text>
          {dsprDriver && (
            <OnCallSwitch
              isLoading={isLoading}
              setDriverOnCallState={setDriverOnCallState}
              dsprDriver={dsprDriver}
            />
          )}
        </View>
      ) : (
        <View style={[styles.container, { backgroundColor: colors.background }]}>
          <Text>Unable to fetch driver data.</Text>
          <Button disabled={isLoading} mode='text' onPress={getDriverData}>
            Try Again
          </Button>
        </View>
      )}
      <StatusBar style='dark' />

      {/*Modal used to prompt users to change location permissions*/}
      <AlertSuccessOrError
          isVisible={showLocationPermissionAlert}
          onDismiss={closeLocationPermissionAlert}
          title={locationPermissionAlertTitle}
          message={locationPermissionAlertText}
          buttonText={'Ok'}
          buttonOnPressSubmit={closeLocationPermissionAlert}
          customAnimationName={'locationPermission'}
          animationViewStyles={{marginTop: 16}}
          titleStyles={{marginTop: 12}}
          messageTextStyles={{marginTop: 16}}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  dsprTitle: {
    fontSize: 22,
    textAlign: 'center',
    paddingTop: 20,
  },
});

export default DashboardDisplay;
