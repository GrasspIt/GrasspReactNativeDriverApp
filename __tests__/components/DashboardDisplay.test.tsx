import React from 'react';
import { render } from '@testing-library/react-native';

import DashboardDisplay from '../../components/DashboardDisplay';

const dspr = 12;
const dsprDriver = {
  id: 12,
  user: 12,
  dspr: 12,
  currentRoute: 12,
  serviceAreas: [1, 2],
};
const isLoading = false;
const getDriverData = jest.fn();
const setDriverOnCallState = jest.fn();

describe('<DashboardDisplay />', () => {
  it('renders correctly', async () => {
    const tree = render(
      <DashboardDisplay
        dspr={dspr}
        dsprDriver={dsprDriver}
        isLoading={isLoading}
        getDriverData={getDriverData}
        setDriverOnCallState={setDriverOnCallState}
      />
    ).toJSON();
    await expect(tree).toMatchSnapshot();
  });
});
