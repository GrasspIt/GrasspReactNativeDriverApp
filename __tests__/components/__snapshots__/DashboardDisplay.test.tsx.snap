// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardDisplay /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f6f6f6",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "paddingTop": 20,
          "textAlign": "center",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "paddingVertical": 12,
        }
      }
    >
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#6200ee"
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        tintColor=""
        value={false}
      />
      <Text>
        Not on Call
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
